### DISCUSSION TITLE
how to prevent pxe boot looping

### DISCUSSION THREAD

**Discussion by wushuzh** [DSpeichert/netbootd#9] [Q&A]
How can I setup the tool to do pxe installation for only one time ?

My server boot order is PXE first, so after installation is done and on next reboot, it may fail into the pxe installation again. And this loop may go forever.

If there is some similiar setting like [cobbler pxe_just_once](https://cobbler.readthedocs.io/en/release28/advanced.html#preventing-boot-loops)

**Comment by DSpeichert** [comment]
Great question!

A method is already built-in, even if it's not obvious. One of the HTTP API features is:
> GET|POST /api/self/suspend-boot
Allows a provisioned host to ask not to be booted again. This does not block DHCP, TFTP or HTTP requests, it only removes NBP information from DHCP responses.

You could have a "late_command" (in Debian-style terms) to curl that endpoint and tell `netbootd` to no longer serve this host.

Something like this:
```
d-i preseed/late_command string curl http://netbootd:8081/api/self/suspend-boot
```

I think it would be good to have a template variable in for the API endpoint but IIRC, there isn't one yet.
